include/seqan3/io/alignment_file/detail.hpp

```diff
@@ -75,128 +73,150 @@ struct view_equality_fn
-                            query_char_type const query_char,
-                            bool const extended_cigar)
 {
-    return (reference_char == gap{})
-                ? (query_char == gap{})
-                    ? 'P'
-                    : 'I'
-                : (query_char == gap{})
-                    ? 'D'
-                    : (extended_cigar)
-                        ? (query_char == reference_char)
-                            ? '='
-                            : 'X'
-                        : 'M';
+    constexpr std::array<char, 6> operators{'M', 'D', 'I', 'P', 'X', '='};  // contains the possible cigar operators.
+    uint8_t key = (static_cast<uint8_t>(reference_char == gap{}) << 1) | static_cast<uint8_t>(query_char == gap{});
+    if (extended_cigar && (key == 0)) // in extended format refine the substitution operator to match/mismatch.
+        key |= ((1 << 2) | static_cast<uint8_t>(query_char == reference_char));  // maps to [4, 5].
+
+    return assign_char_to(operators[key], cigar_op{});
 }
 
-/*!\brief Transforms an alignment represented by two aligned sequences into the
- *        corresponding CIGAR string.
+/*!\brief Creates a CIGAR string (SAM format) given an alignment represented by two aligned sequences.
  * \ingroup alignment_file
  *
- * \tparam ref_seq_type    Must model std::ranges::forward_range. The value_type must
- *                         be equality comparable to seqan3::gap.
- * \tparam query_seq_type  Must model std::ranges::forward_range. The value_type must
- *                         be equality comparable to seqan3::gap.
- * \param  ref_seq         The reference sequence to compare against the query sequence.
- * \param  query_seq       The query sequence to build the CIGAR string for.
+ * \tparam alignment_type  Must model the seqan3::tuple_like and must have std::tuple_size 2.
+ *                         Each tuple element must be a range over values comparable to seqan3::gap.
+ * \param  alignment       The alignment, represented by a pair of aligned sequences,
+ *                         to be transformed into CIGAR_vector based on the
+ *                         second (query) sequence.
  * \param  query_start_pos The start position of the alignment in the query
  *                         sequence indicating soft-clipping.
  * \param  query_end_pos   The end position of the alignment in the query
  *                         sequence indicating soft-clipping.
  * \param  extended_cigar  Whether to print the extended cigar alphabet or not. See CIGAR operation.
- * \returns An std::string representing the alignment as a CIGAR string.
+ * \returns An std::vector<seqan3::cigar> representing the alignment.
  *
- * ### Theoretical Example:
+ * \details
  *
- * The following alignment reference sequence on top and the query sequence at
- * the bottom.
+ * \attention Note that CIGAR elements (respectively by their CIGAR operation) are always related to one of the two
```

Comment:
```suggestion
The first sequence is always considered the reference sequence while the second one is considered the query sequence. The cigar operations regarding insertions and deletions are set accordingly.
```

Is it this, what you want to express?
If yes, could make the same/similar note above for the other function?